ext/mixed/js/core.js

```diff
@@ -250,15 +250,35 @@ class EventListenerCollection {
         return this._eventListeners.length;
     }
 
-    addEventListener(node, type, listener, options) {
-        node.addEventListener(type, listener, options);
-        this._eventListeners.push([node, type, listener, options]);
+    addEventListener(object, ...args) {
+        object.addEventListener(...args);
+        this._eventListeners.push(['removeEventListener', object, ...args]);
+    }
+
+    addListener(object, ...args) {
+        object.addListener(args);
+        this._eventListeners.push(['addListener', object, args]);
+    }
+
+    on(object, ...args) {
+        object.on(args);
+        this._eventListeners.push(['off', object, args]);
     }
 
     removeAllEventListeners() {
         if (this._eventListeners.length === 0) { return; }
-        for (const [node, type, listener, options] of this._eventListeners) {
-            node.removeEventListener(type, listener, options);
+        for (const [type, object, ...args] of this._eventListeners) {
```

Comment:
`type` might not be the best name for the name of the function to remove the event listener. Unless they have some established names, could just go with `removeFunctionName` or `addFunctionName`.

Reply:
a8b39b7e1726c288f52fdc7d9255803de33ce7e3